### PR TITLE
Use enum instead of int in virtual methods return type

### PR DESCRIPTION
**Pull request by raulsntos**
- In class `ResourceFormatLoader`:
	- Change return type of `_rename_dependencies` method to `Error`.
- In class `ResourceFormatSaver`:
	- Change return type of `_save` method to `Error`.
- In class `EditorImportPlugin`:
	- Change return type of `_import` method to `Error`.
- In class `EditorInspectorPlugin`:
	- Change type of `usage_flags` parameter in `_parse_property` method to `PropertyUsageFlags`.
- In class `EditorSceneFormatImporter`:
	- Change return type of `_get_import_flags` method to `uint32_t`.
- In class `GLTFDocumentExtension`:
	- Change return type of `_import_preflight` method to `Error`.
	- Change return type of `_parse_node_extensions` method to `Error`.
	- Change return type of `_import_post_parse` method to `Error`.
	- Change return type of `_import_node` method to `Error`.
	- Change return type of `_import_post` method to `Error`.
	- Change return type of `_export_preflight` method to `Error`.
	- Change return type of `_export_node` method to `Error`.
	- Change return type of `_export_post` method to `Error`.
- In class `MultiplayerAPI`:
	- Change return type of `_poll` method to `Error`.
	- Change return type of `_rpc` method to `Error`.
	- Change return type of `_object_configuration_add` method to `Error`.
	- Change return type of `_object_configuration_remove` method to `Error`.
- In class `VisualShaderNodeCustom`:
	- Change return type of `_get_return_icon_type` method to `PortType`.
	- Change return type of `_get_input_port_type` method to `PortType`.
	- Change return type of `_get_output_port_type` method to `PortType`.
- In class `XRInterfaceExtension`:
	- Change return type of `_get_tracking_status` method to `XRInterface::TrackingStatus`.
	- Change return type of `_get_play_area_mode` method to `XRInterface::PlayAreaMode`.
	- Change parameter type of `_set_play_area_mode` method to `XRInterface::PlayAreaMode`.

### CODE REVIEW

File: scene/resources/visual_shader.h [hunk 1]
```diff
@@ -371,10 +371,10 @@ class VisualShaderNodeCustom : public VisualShaderNode {
 	GDVIRTUAL0RC(String, _get_category)
 	GDVIRTUAL0RC(int, _get_return_icon_type)
```
Comment (raulsntos) [inline scene/resources/visual_shader.h:372]: Where is `_get_return_icon_type` used? I couldn't find a `GDVIRTUAL_CALL`. It seems like the return type should also be `PortType` but I'm not sure so I didn't change it.

Comment (akien-mga) [reply scene/resources/visual_shader.h:372]: It's used here:

https://github.com/godotengine/godot/blob/d2d5e548c9cac48a89b97ed4dd9bfc93b4528f0d/editor/plugins/visual_shader_editor_plugin.cpp#L1244

Could possibly be changed to `GDVIRTUAL_CALL`?

File: core/io/resource_loader.cpp [hunk 2]
```diff
@@ -170,7 +170,7 @@ Error ResourceFormatLoader::rename_dependencies(const String &p_path, const Hash
 		deps_dict[E.key] = E.value;
 	}
 
-	int64_t err = OK;
+	Error err = OK;
 	GDVIRTUAL_CALL(_rename_dependencies, p_path, deps_dict, err);
 	return (Error)err;
```
Comment (akien-mga) [inline core/io/resource_loader.cpp:175]: The cast could maybe be removed?

**Review by akien-mga — COMMENTED** [review]
Looks good to me, provided that it actually works for both GDScript and GDExtension. It looks like this was intentionally written with int initially, probably because enums weren't supported via this interface.

File: scene/resources/visual_shader.h (hunk 1 above)
Comment (raulsntos) [reply scene/resources/visual_shader.h:372]: I don't think we can use the `GDVIRTUAL_CALL` macro here. The method is called on `p_custom_node` and is defined as `protected` in a different type so likely not accessible anyway.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**